### PR TITLE
Update win2lin scripts to match production

### PR DESCRIPTION
**Pull request by lowenna**
Fixes https://github.com/docker/docker/issues/23379. Updating the scripts in the docker repo with the current production Jenkins versions.

@justincormack 

Signed-off-by: John Howard jhoward@microsoft.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @tianon

File: hack/Jenkins/W2L/setup.sh [hunk 1]
```diff
@@ -119,31 +129,32 @@ fi
...
-
 	# This SHOULD never happen, but just in case, also blow away any containers
 	# that might be around.
-	! if [ ! `docker ps -aq | wc -l` -eq 0 ]; then
+	! if [ ! $(docker ps -aq | wc -l) -eq 0 ]; then
 		echo WARN: There were some leftover containers. Cleaning them up.
 		! docker rm -f $(docker ps -aq)
 	fi
+	
+    # Force remove the image if it exists
```
Comment (thaJeztah) [inline hack/Jenkins/W2L/setup.sh]: nit: this is using spaces, not tabs

File: hack/Jenkins/W2L/setup.sh [hunk 2]
```diff
@@ -184,24 +213,25 @@ if [ $ec -eq 0 ]; then
 	echo "INFO: Starting build of a Linux daemon to test against, and starting it..."
 	set -x
 	# aufs in aufs is faster than vfs in aufs
-	docker run $run_extra_args -e DOCKER_GRAPHDRIVER=aufs --pid host --privileged -d --name "docker-$COMMITHASH" --net host "docker:$COMMITHASH" bash -c "echo 'INFO: Compiling' && date && hack/make.sh binary && echo 'INFO: Compile complete' && date && cp bundles/$(cat VERSION)/binary/docker /bin/docker && echo 'INFO: Starting daemon' && exec docker daemon -D -H tcp://0.0.0.0:$port_inner $daemon_extra_args"
+	docker run -d $run_extra_args -e DOCKER_GRAPHDRIVER=aufs --pid host --privileged --name "docker-$COMMITHASH" --net host "docker:$COMMITHASH"
 	ec=$?
 	set +x
 	if [ 0 -ne $ec ]; then
-		echo "ERROR: Failed to compile and start the linux daemon"
+	    	echo "ERROR: Failed to compile and start the linux daemon"
```
Comment (thaJeztah) [inline hack/Jenkins/W2L/setup.sh]: nit: mixed tabs/spaces

File: hack/Jenkins/W2L/setup.sh [hunk 3]
```diff
@@ -184,24 +213,25 @@ if [ $ec -eq 0 ]; then
...
 fi
 
 # Build locally.
 if [ $ec -eq 0 ]; then
 	echo "INFO: Starting local build of Windows binary..."
 	set -x
-	export TIMEOUT="5m"
+	export TIMEOUT="120m"
 	export DOCKER_HOST="tcp://$ip:$port_inner"
+    # This can be removed
```
Comment (thaJeztah) [inline hack/Jenkins/W2L/setup.sh]: ...

File: hack/Jenkins/W2L/setup.sh [hunk 4]
```diff
@@ -221,19 +255,20 @@ if [ $ec -eq 0 ]; then
...
 	hack/make.sh test-integration-cli
 	ec=$?
 	set +x
 	if [ 0 -ne $ec ]; then
 		echo "ERROR: CLI test failed."
 		# Next line is useful, but very long winded if included
-		# docker -H=$MAIN_DOCKER_HOST logs "docker-$COMMITHASH"
-	fi
+		docker -H=$MAIN_DOCKER_HOST logs --tail 100 "docker-$COMMITHASH"
+    fi
```
Comment (thaJeztah) [inline hack/Jenkins/W2L/setup.sh]: spaces -> tabs

**Comment by justincormack** [comment]
Thanks!

**Comment by jhowardmsft** [comment]
To be clear - this is exactly "as is" in production. This is only in the docker repo to keep a copy of it....

**Comment by justincormack** [comment]
Yes, we should just merge as is, and use a different PR to change it.

**Comment by thaJeztah** [comment]
sgtm

**Comment by crosbymichael** [comment]
LGTM

**Comment by vdemeester** [comment]
All green.
LGTM .🐺

**Pull request merged**